### DISCUSSION TITLE
Add custom library to vcpkg based on given .dll .lib and .h

### DISCUSSION THREAD

**Discussion by le1nax** [microsoft/vcpkg#34509] [Q&A] [answered]
Hello,

from my understanding, to add a custom library to vcpkg (package is not included in default vcpkg ports, provider is external, provided .h .dll and .lib), I have to follow these steps:

1. Create a port in ../vcpkg/ports/mypackage
2. In there, create a portfile.cmake and copy .h .dll .lib
3. Create vcpkg.json
4. Create mypackageConfig.cmake
5. Create 'CONTROL' file
6. Do `vcpkg install` (inside the port dir)
7. Integrate package into cmake project with find_package(myPackage CONFIG REQUIRED) and target_link_libraries(myTarget PRIVATE myPackage )

When Building with mingw gcc, it returns error: ../x86_64-w64-mingw32/bin/ld.exe: cannot find -lpcanbasic: No such file or directory (pcanBasic being myPackage in this case). 

It appears to me, that the package was not properly installed. What I found odd is, that when initally doing `vcpkg install` in the port directory, it returned "All requested packages are currently installed. Restored 0 package(s) from archives". I assume I have mistaken the way of installing the custom port. I thought doing `vcpkg install` inside the port directory installs the custom library specified in the portfile, but it seems it rather installs the dependencies defined in the vcpkg.json. This would explain the return statement.
How do I actually install the package? Is my procedure of adding the custom library correct? What am I missing here?

**Comment by Osyotr** [comment, marked as answer]
1-3 correct
4 IMHO if a package does not natively provide cmake config you should write find-module in your project (though for vendored libs this requirement is optional)
5 Wrong. You already created manifest file in (3)
6 Wrong. That way you're installing dependencies of the package, not the package itself. Use `vcpkg install mypackage` instead.
7 ok

**Reply by le1nax** [reply]
Thanks that made things alot clearer. 
Yet I got confused with installing the packet itself with `vcpkg install mypackage` in vcpkg root dir, because it says this was only possible in classic mode. Having manifest mode enabled, mypackage gets installed the moment I install another project which depends on mypackage it seems.
